force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +137,19 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
+        String concatenatedStr = '';
 
         // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
+        for (String name : scientistNames) {
+            if (name.toLowerCase().contains('g')) {
+                concatenatedStr += name + ',';
```

Comment:
You can use String.join on a List of scientistNamesWithG so that you don't have to call removeEnd or store them as a CSV before you're ready to generate the CSV.